distributed/deploy/adaptive.py

```diff
@@ -80,13 +132,13 @@ def _adapt(self):
         self._adapting = True
         try:
             if self.should_scale_up():
```

Comment:
Should probe should_scale_up/down before calling any adaption functions? I know that the actions are not immediate, but I could imagine scaling up here and immediately scaling down a few lines later.